stytch/cryptowallet/cryptowallet.go

```diff
@@ -48,3 +49,46 @@ func (c *Client) Authenticate(body *stytch.CryptoWalletAuthenticateParams,
 	err = c.C.NewRequest("POST", path, nil, jsonBody, &retVal)
 	return &retVal, err
 }
+
+func (c *Client) AuthenticateWithClaims(
+	body *stytch.CryptoWalletAuthenticateParams,
+	claims interface{},
+) (*stytch.CryptoWalletAuthenticateResponse, error) {
+	path := "/crypto_wallets/authenticate"
+
+	var jsonBody []byte
+	var err error
+	if body != nil {
+		jsonBody, err = json.Marshal(body)
+		if err != nil {
+			return nil, stytcherror.NewClientLibraryError("Oops, something seems to have gone wrong " +
+				"marshalling the authenticate request body")
+		}
+	}
+
+	b, err := c.C.RawRequest("POST", path, nil, jsonBody)
+	if err != nil {
+		return nil, err
+	}
+
+	// First extract the Stytch data.
+	var retVal stytch.CryptoWalletAuthenticateResponse
+	if err := json.Unmarshal(b, &retVal); err != nil {
+		return nil, fmt.Errorf("unmarshal CryptoWalletAuthenticateResponse: %w", err)
+	}
+
+	// Then extract the custom claims. Build a claims wrapper using the caller's `claims` value so
+	// the unmarshal fills it.
+	wrapper := stytch.ClaimsWrapper{
+		Session: struct {
+			Claims interface{} `json:"custom_claims"`
+		}{
+			Claims: claims,
+		},
```

Comment:
I think we can clean up some of this awkwardness by naming the inner struct type too:
```go
Session: stytch.SessionClaims{ // or whatever you want to call it
	Claims: claims,
},
```

Reply:
https://github.com/stytchauth/stytch-go/pull/77/commits/dfbb30b779e6f147f8826be224139e93d3315407